src/app/app.module.ts

```diff
@@ -1,14 +1,28 @@
 import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { AppComponent } from '../app/app.component/app.component';
+import { DashboardComponent } from '../app/dashboard.component/todo.dashboard';
+import { EditComponent } from '../app/edit.component/edit.component';
 
-import { AppComponent } from './app.component';
+import{RouterModule} from '@angular/router';
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    DashboardComponent,
+    EditComponent
   ],
   imports: [
-    BrowserModule
+    BrowserModule,
+    RouterModule.forRoot([
+      {path: '', pathMatch: 'full', redirectTo: 'dashboard'},
+      {path:'',component:DashboardComponent},
+      {path:'edit',component:EditComponent},
+      {path:'Edit/:id',component:EditComponent}
```

Comment:
why 2 route for single component.

Reply:
sir when on add button i navigate it give error and i used to single route it don't give error thats why i used it